### DISCUSSION TITLE
parse braces with text

### DISCUSSION THREAD

**Discussion by joycey97** [earwig/mwparserfromhell#330] [Q&A]
what is the function or set of operations to parse from Infobox line:
```| foundation = founded {{start date and age|September 25, 1928}} as [[ABC Company]]```
to plain text
```founded September 25, 1928 as ABC Company```

also, from:
```revenue = {{increase}} {{USD|9.98 billion|link=yes}} (2023)```
to
```US$9.98 billion (2023)``` (as displayed on Wikipedia page)

**Comment by JJMC89** [comment]
The process is like I demonstrated in #328. You have to know what the templates to to their inputs.

Depending on your goals, you might be better off using the [`parse`](https://www.mediawiki.org/wiki/API:Parsing_wikitext) or [`expandtemplates`](https://www.mediawiki.org/wiki/API:Expandtemplates) APIs.

**Comment by earwig** [comment]
First one can be done with a single call:

```python
>>> mwparserfromhell.parse('founded {{start date and age|September 25, 1928}} as [[ABC Company]]').strip_code(keep_template_params=True)
'founded September 25, 1928 as ABC Company'
```

Second one, as JJMC89 says, you need to know a bit about what the template actually produces to get it exactly right. mwparserfromhell just parses the wikitext, it doesn't have a way to solve this problem for you directly.
